src/core/util/ref_counted_ptr.h

```diff
@@ -439,6 +439,31 @@ struct RefCountedPtrEq {
   }
 };
 
+// Heterogenous lookup support.
+template <typename T>
+struct WeakRefCountedPtrHash {
```

Comment:
Do we need these?  The existing `RefCountedPtrHash` and `RefCountedPtrEq` also support `WeakRefCountedPtr<>`.

Reply:
This avoids accidental comparison with RefCountedPtr... and increases the firewall against accidentally mixing those spaces.